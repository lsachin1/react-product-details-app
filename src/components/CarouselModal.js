import React from 'react';
import Modal from 'react-modal'
import "react-responsive-carousel/lib/styles/carousel.min.css";
import { Carousel } from 'react-responsive-carousel';

export const OpenModal = (props) => {
	let carouselArr = [];

	const customStyles = {
		  content : {
		    top                   : "50%",
		    left                  : "50%",
		    right                 : "auto",
		    bottom                : "auto",
		    marginRight           : "-50%",
		    transform             : "translate(-50%, -50%)"
		  }
	};

	carouselArr = props.selectedThumbail.map((image, index) => {
		return (
			<div key={index}>
				<img src={image.href} className="carouselImage" width={image.width+"px"} height={image.height+"px"} alt={image.alt}/>
			</div>
		)
	});

	return (
		<Modal
			isOpen={props.isModalOpened}
			onRequestClose={props.handleClearModalOpened}
			style = {customStyles}
			contentLabel="Selected Image"
		>
			<span className="close" onClick={props.handleClearModalOpened}>X</span>
			<Carousel>
				{carouselArr}
            </Carousel>

		</Modal>
	)
}