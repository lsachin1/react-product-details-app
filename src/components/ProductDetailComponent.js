import React from 'react';

export const ProductDetailComponent = ({url, name, regularLow, regularHigh, specialLow, specialHigh, type}) => {
	
	let specialType = type.charAt(0).toUpperCase() + type.slice(1)

	return (
		<div className="productDeatilContainer">
			<p>
				<a href={url} target="_blank" rel="noopener noreferrer">
					{name}
				</a>
			</p>
			<p>
				<span><s>${regularLow} - ${regularHigh}</s></span>
				<span> {specialType} ${specialLow} - ${specialHigh}</span>
			</p>
		</div>
	)
}
