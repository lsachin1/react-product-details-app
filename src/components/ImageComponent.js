import React from 'react';
import { OpenModal } from './CarouselModal';
import { useState } from 'react';

export const ImageComponent = ({url, width, height, thumbImageArr, alt}) => {
	let openModal = "";
	const [isModalOpened, setModalOpened] = useState(false)
	const [selectedThumbail] = useState(thumbImageArr)

	const handleClearModalOpened = () => {
		setModalOpened(false)
	};

	if(thumbImageArr.length > 0){
		openModal = 
			<OpenModal 
				isModalOpened={isModalOpened}
				selectedThumbail={selectedThumbail}
				handleClearModalOpened={handleClearModalOpened}
			/>
		
	}
	return (
		<div className="imageContainer">
			<a onClick={() => setModalOpened(!isModalOpened)} href="javascript:void()">
				<img src={url} width={width+"px"} height={height+"px"} alt={alt}/>
			</a>
			{openModal}
			
		</div>
	)
}