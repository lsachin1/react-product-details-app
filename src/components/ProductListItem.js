import React from 'react';
import { useState } from 'react';
import { OpenModal } from './CarouselModal';

export const ProductListItem = (productItemValues) => {
	
	const [isModalOpened, setModalOpened] = useState(false);
	const [selectedThumbail, setSelectedThumbnail] = useState(productItemValues.images);

	const handleClearModalOpened = () => {
		setModalOpened(false)
	}

	let type = productItemValues.priceRange.type.charAt(0).toUpperCase() + productItemValues.priceRange.type.slice(1);
	return (
		
			<div className="productItemContainer">
				<div className="productIteamInner">
					<div className="imageContainer">
						<a onClick={() => setModalOpened(!isModalOpened)}>
							<img src={productItemValues.hero.href} width={productItemValues.hero.width+"px"} height={productItemValues.hero.height+"px"}/>
						</a>
					</div>
					<div className="productDeatilContainer">
						<p>{productItemValues.name}</p>
						<span><s>${productItemValues.priceRange.regular.low} - ${productItemValues.priceRange.regular.high}</s></span>
						<span> {type} ${productItemValues.priceRange.selling.low} - ${productItemValues.priceRange.selling.high}</span>
					</div>
				</div>
				<OpenModal isModalOpened={isModalOpened} selectedThumbail={selectedThumbail} handleClearModalOpened={handleClearModalOpened}/>
			</div>
			
		
	)
}