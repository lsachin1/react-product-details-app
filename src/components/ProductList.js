import React from 'react';
import { ProductListItem }  from './ProductListItem';

export const ProductList = (props) => {
	let count = props.data.length;
	let content = [];
	if(count > 0) {
		const rows = [...Array( Math.ceil(props.data.length / 2) )];
		const productRows = rows.map((row, index) => {
			return props.data.slice(index * 2 , index * 2 + 2)
		});
		content = productRows.map((row, idx) => {
			return (
				<div className="productContainer" key={idx}>
					{
						row.map((product, index) => {
							return <ProductListItem key={product.id} {...product} ind={index}/>
						})
					}
				</div>
			)
		})
	}


	return (
        <div>
          {content.length > 0 ? content: 'No data'}
        </div>
    );
}
