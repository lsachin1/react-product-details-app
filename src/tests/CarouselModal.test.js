import React from 'react';
import ReactDOM from 'react-dom';
import ReactShallowRenderer from 'react-test-renderer/shallow';
import { OpenModal } from '../components/CarouselModal';

const thumbnailArr = [
	{
		size: "m",
		meta: "",
		alt: "",
		rel: "althero",
		width: 363,
		href: "https://www.westelm.com/weimgs/ab/images/wcm/products/201914/0161/organic-watercolor-wavelines-duvet-cover-shams-3-m.jpg",
		height: 363
	},
	{
		size: "m",
		meta: "",
		alt: "",
		rel: "althero",
		width: 363,
		href: "https://www.westelm.com/weimgs/ab/images/wcm/products/201916/0026/organic-watercolor-wavelines-duvet-cover-shams-m.jpg",
		height: 363
	},
	{
		size: "m",
		meta: "",
		alt: "",
		rel: "althero",
		width: 363,
		href: "https://www.westelm.com/weimgs/ab/images/wcm/products/201916/0026/organic-watercolor-wavelines-duvet-cover-shams-1-m.jpg",
		height: 363
	}
];


it('render the Modal with thumbnail images', () => {
	const renderer = new ReactShallowRenderer();
	renderer.render(<OpenModal selectedThumbail={thumbnailArr} />);
	expect(renderer.getRenderOutput()).toMatchSnapshot();
})
