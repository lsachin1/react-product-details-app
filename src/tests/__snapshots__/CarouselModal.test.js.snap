// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render the Modal with thumbnail images 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  closeTimeoutMS={0}
  contentLabel="Selected Image"
  isOpen={false}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnOverlayClick={true}
  style={
    Object {
      "content": Object {
        "bottom": "auto",
        "left": "50%",
        "marginRight": "-50%",
        "right": "auto",
        "top": "50%",
        "transform": "translate(-50%, -50%)",
      },
    }
  }
>
  <span
    className="close"
  >
    X
  </span>
  <Carousel
    autoPlay={false}
    axis="horizontal"
    centerMode={false}
    centerSlidePercentage={80}
    dynamicHeight={false}
    emulateTouch={false}
    infiniteLoop={false}
    interval={3000}
    onChange={[Function]}
    onClickItem={[Function]}
    onClickThumb={[Function]}
    selectedItem={0}
    showArrows={true}
    showIndicators={true}
    showStatus={true}
    showThumbs={true}
    statusFormatter={[Function]}
    stopOnHover={true}
    swipeScrollTolerance={5}
    swipeable={true}
    transitionTime={350}
    useKeyboardArrows={false}
    verticalSwipe="standard"
    width="100%"
  >
    <div>
      <img
        alt=""
        className="carouselImage"
        height="363px"
        src="https://www.westelm.com/weimgs/ab/images/wcm/products/201914/0161/organic-watercolor-wavelines-duvet-cover-shams-3-m.jpg"
        width="363px"
      />
    </div>
    <div>
      <img
        alt=""
        className="carouselImage"
        height="363px"
        src="https://www.westelm.com/weimgs/ab/images/wcm/products/201916/0026/organic-watercolor-wavelines-duvet-cover-shams-m.jpg"
        width="363px"
      />
    </div>
    <div>
      <img
        alt=""
        className="carouselImage"
        height="363px"
        src="https://www.westelm.com/weimgs/ab/images/wcm/products/201916/0026/organic-watercolor-wavelines-duvet-cover-shams-1-m.jpg"
        width="363px"
      />
    </div>
  </Carousel>
</Modal>
`;
